databases/week3/Homework_queries.sql

```diff
@@ -0,0 +1,330 @@
+DELETE FROM
+  `Review`
+WHERE
+  Review.id = 2;
+
+-- Additional queries
+-- Now add a couple of different meals, reservations and reviews with different attributes. With those meals create the following queries
+INSERT INTO
+  Meal (
+    id,
+    title,
+    description,
+    location,
+    Meal.when,
+    price,
+    max_reservations,
+    created_date
+  )
+VALUES
+  (
+    1,
+    'Peking Duck',
+    'crunchy duck piles with sauce',
+    'Copenhagen',
+    '2023-01-28 00:00:00',
+    500.0,
+    6,
+    '2023-01-24'
+  );
+
+INSERT INTO
+  `Meal` (
+    id,
+    title,
+    location,
+    description,
+    Meal.when,
+    max_reservations,
+    created_date,
+    price
+  )
+VALUES
+  (
+    1,
+    'Noodles',
+    'CPH',
+    'long lines of boiled dough',
+    '2023-03-03 17:00:00',
+    100,
+    '2023-01-24',
+    100
+  );
+
+INSERT INTO
+  `Review` (
+    id,
+    title,
+    description,
+    stars,
+    created_date,
+    meal_id
+  )
+VALUES
+  (
+    3,
+    'Grinch',
+    'Not tasty at all',
+    1,
+    '2023-01-26',
+    3
+  );
+
+INSERT INTO
+  Reservation (
+    id,
+    number_of_guests,
+    meal_id,
+    created_date,
+    contact_phonenumber,
+    contact_name,
+    contact_email
+  )
+VALUES
+  (
+    3,
+    5,
+    2,
+    '2023-01-29',
+    '11-11-11-00',
+    'Mickey Mouse',
+    'main@disney.com'
+  );
+
+-- Functionality
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT
+  *
+FROM
+  `Meal`
+WHERE
+  Meal.price <= 200;
+
+-- Get meals that still has available reservations
+SELECT
+  *
+FROM
+  `Meal`
+WHERE
+  Meal.max_reservations >= COALESCE(
+    (
+      SELECT
+        sum(Reservation.number_of_guests)
+      FROM
+        `Reservation`
+      where
+        Reservation.meal_id = Meal.id
+    ),
+    0
+  );
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT
+  *
+from
+  `Meal`
+where
+  Meal.title like '%rød grød med%';
+
+-- Get meals that has been created between two dates
+SELECT
+  *
+FROM
+  `Meal`
+where
+  Meal.created_date BETWEEN '2023-01-01' AND '2023-01-24';
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT
+  *
+FROM
+  `Meal`
+limit
+  2;
+
+-- Get the meals that have good reviews
+SELECT
+  Meal.title,
+  AVG(Review.stars)
+FROM
+  `Meal`
+  join `Review` on Meal.id = Review.meal_id
+GROUP BY
+  Meal.title
+HAVING
+  AVG(Review.stars) > 4.2;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT
+  Reservation.contact_name,
+  Reservation.number_of_guests,
+  Meal.title,
+  Meal.when
+FROM
+  `Reservation`
+  JOIN `Meal` on Reservation.meal_id = Meal.id
+WHERE
+  Meal.title like '%duck%'
+ORDER BY
+  Reservation.created_date DESC;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT
+  Meal.title,
+  AVG(Review.stars)
+FROM
+  `Meal`
+  join `Review` on Meal.id = Review.meal_id
+GROUP BY
+  Meal.title
+ORDER BY
+  AVG(Review.stars) DESC;
```

Comment:
Nothing wrong with the script, but why such many lines of code (330 lines) for a small task?

Please make your code of line as short & clean as possible for better readability- https://www.dpriver.com/pp/sqlformat.htm
For instance, this can be like below 
```
SELECT  Meal.title,  AVG(Review.stars)
FROM  `Meal`
join `Review` on Meal.id = Review.meal_id
GROUP BY  Meal.title
ORDER BY  AVG(Review.stars) DESC;
```

Reply:
Thanks for the comment! I use the MySQL v6.1.3 extension for VSCode by Weijan Chen. And somehow it lints all the queries like so... Probably there is a config somewhere, but I couldn't find it :(